metasyn/metaframe.py

```diff
@@ -473,6 +488,18 @@ def synthesize(self, n: Optional[int] = None, seed: Optional[int] = None) -> pl.
         synth_dict = {var.name: var.draw_series(n, seed=None) for var in self.meta_vars}
         return pl.DataFrame(synth_dict)
 
+    def write_synthetic(self, file_name: Union[None, Path, str] = None,
+                        n: Optional[int] = None, seed: Optional[int] = None,
+                        file_format: Union[None, dict, BaseFileHandler] = None):
+        syn_df = self.synthesize(n, seed)
+        if file_format is not None:
+            self.file_format = file_format
+        if self.file_format is None:
+            raise ValueError("Cannot write synthetic dataset without file handler."
+                             " Use write_synthetic(..., file_handler=your_file_handler)")
```

Comment:
Check for file format before synthesizing

```suggestion
        if file_format is not None:
            self.file_format = file_format
        if self.file_format is None:
            raise ValueError("Cannot write synthetic dataset without file handler."
                             " Use write_synthetic(..., file_handler=your_file_handler)")
        syn_df = self.synthesize(n, seed)
```